sources/core/Stride.Core.Reflection/TypeDescriptors/PrimitiveDescriptor.cs

```diff
@@ -12,14 +12,15 @@ namespace Stride.Core.Reflection
     /// </summary>
     public class PrimitiveDescriptor : ObjectDescriptor
     {
-        private static readonly List<IMemberDescriptor> EmptyMembers = new List<IMemberDescriptor>();
-        private readonly Dictionary<string, object> enumRemap;
+        private static readonly List<IMemberDescriptor> EmptyMembers = [];
+        private readonly Dictionary<string, object?> enumRemap;
 
         public PrimitiveDescriptor(ITypeDescriptorFactory factory, Type type, bool emitDefaultValues, IMemberNamingConvention namingConvention)
             : base(factory, type, emitDefaultValues, namingConvention)
         {
             if (!IsPrimitive(type))
                 throw new ArgumentException("Type [{0}] is not a primitive");
+            enumRemap = new Dictionary<string, object?>(StringComparer.OrdinalIgnoreCase);
```

Comment:
Looking at the rest of the logic, it looks like you can rollback this change and mark `enumRemap` as nullable ?

Reply:
i thought,

if something is found enumRemap is set in any case
if nothing is found its an empty dictionary which is also valid for all the usages in the class

so later on the null checks can be removed which i forgot, 

its like returning an empty list instead of returning null to be valid in any case